### PR TITLE
webdojo: slogs

### PR DESCRIPTION
**Pull request by Fang-**
Connects to the SSE stream on `/~/slog` to receive printf output from the
runtime, and pretends those are incoming `%txt` effects.

Depends on #3557, which introduces the endpoint used here. Probably shouldn't be merged until that has gone out in a vere release? But this fails gracefully, so maybe we can pass on juggling pending PRs.

### CODE REVIEW

File: pkg/interface/src/views/apps/dojo/subscription.js
Comment (matildepark) [inline pkg/interface/src/views/apps/dojo/subscription.js:37]: Makes webdojo itself messier than necessary honestly. This should be a `console.log`.

**Comment by matildepark** [comment]
I feel weird approving stuff that is awaiting next-sys changes. The "PR is blocked on a next-sys release" case isn't well covered IMO.

**Comment by Fang-** [comment]
> But this fails gracefully, so maybe we can pass on juggling pending PRs.

With the requested change, this even fails _silently_. Unless the user is looking at browser console.

In this particular case I don't think it's a big deal to merge this ahead of the runtime-side dependency. But it's your call!

**Comment by matildepark** [comment]
slog reconnects every few seconds ... 

image

**Comment by Fang-** [comment]
Oops, looks like I threw a baby out with the bathwater in b12f892. Will repair!

**Comment by philipcmonk** [comment]
> The "PR is blocked on a next-sys release" case isn't well covered IMO.

Without following the full context, generally that just means it should be PR'd against next-sys.  Of course it'll look a little different once landscape and arvo are fully decoupled.

**Comment by matildepark** [comment]
I generally agree. Once the two are decoupled, though, we are essentially pinned to an Arvo release (maybe a Git submodule? Idk) so we’d have to think about changes with dependencies. Perhaps just strongly versioning.

**Comment by philipcmonk** [comment]
Yeah I think you'll have a branch that depends on future arvo changes which is separate from the one which doesn't.  That branch can even have changes which are *required* to make landscape work with the new version -- we can still unidirectionally couple by having anyone distributing landscape (eg ~zod) only send the new arvo changes with the corresponding landscape changes.  Other software distributors should do the same thing.

**Comment by Fang-** [comment]
@matildepark this fails more gracefully now. Please take another look. Afaic this can just slide into the next-userspace queue, but your call on when/where to merge this.

**Comment by matildepark** [comment]
Whoops. Will look tn.

**Review by matildepark — APPROVED** [review]
Fails gracefully.

Incidentally this makes Dojo overflow scroll horizontally, which I guess was on purpose.

Before / after:

image


image

**Pull request merged**
